src/resolvers/index.js

```diff
@@ -0,0 +1,21 @@
+'use strict'
+
+const errCode = require('err-code')
+
+const resolvers = {
+  'dag-pb': require('./unixfs-v1'),
+  raw: require('./raw'),
+  'dag-cbor': require('./dag-cbor')
+}
+
+const resolve = (cid, name, path, toResolve, ipld) => {
+  const resolver = resolvers[cid.codec]
+
+  if (!resolver) {
+    throw errCode(new Error(`No resolver for codec ${cid.codec}`), 'ENORESOLVER')
+  }
+
+  return resolver(cid, name, path, toResolve, resolve, ipld)
+}
```

Comment:
Counter argument to not using `async` keyword for async APIs is that callers of this code cannot use `.catch`:

```js
const ethCid = new CID('z43AaGEvwdfzjrCZ3Sq7DKxdDHrwoaPQDtqF4jfdkNEVTiqGVFW')
resolve(ethCid /*, ... */).catch(console.error) // uncaught exception
```